### ISSUE TITLE
[PPC] `rationalize` on Float16 returns wrong results.

### ISSUE THREAD

**Issue by vchuravy** [JuliaLang/julia#38637]
Julia 1.5.2:
```
julia> rationalize(Float16(1.0))
1//1

julia> rationalize(Float16(0.1))
1//10
```

Julia master:
```
julia> rationalize(Float16(1.0))
0//1

julia> rationalize(Float16(0.1))
1//9
```

**Comment by JeffBezanson** [comment]
What should the result be?

**Comment by vchuravy** [comment]
> What should the result be?

I mistakenly copied the wrong output, fixed now.

**Comment by vchuravy** [comment]
Seemingly fixed on 8dd959efb8
